texar/data/data/multi_aligned_data.py

```diff
@@ -475,166 +484,43 @@ def process(self, raw_example: Tuple[Union[str, Dict[str, Any]], ...]) \
-                if (max_seq_length is not None and
-                        len(raw_tokens) > max_seq_length):
-                    if dataset_hparams[i]["length_filter_mode"] is \
-                            _LengthFilterMode.TRUNC.value:
-                        raw_tokens = raw_tokens[:max_seq_length]
-
-                if (dataset_hparams[i].bos_token is not None and
-                        dataset_hparams[i].bos_token != ''):
-                    raw_tokens.insert(0, dataset_hparams[i].bos_token)
-                if (dataset_hparams[i].eos_token is not None and
-                        dataset_hparams[i].eos_token != ''):
-                    raw_tokens.append(dataset_hparams[i].eos_token)
-                transforms = self._other_transforms[i]
-                if len(transforms) > 0:
-                    for transform in transforms:
-                        raw_tokens = transform(raw_tokens)
-                processed_examples.append(raw_tokens)
+                example_i = self._databases[i].process(raw_example_i)
             elif _is_scalar_data(dataset_hparams[i]["data_type"]):
-                scalar_ex: Union[str, int, float] = str(raw_example_i)
-                data_type = dataset_hparams[i]["data_type"]
-                if data_type == "int":
-                    scalar_ex = int(scalar_ex)
-                elif data_type == "float":
-                    scalar_ex = float(scalar_ex)
-                transforms = self._other_transforms[i]
-                if len(transforms) > 0:
-                    for transform in transforms:
-                        scalar_ex = transform(scalar_ex)
-                processed_examples.append(scalar_ex)
+                example_i = self._databases[i].process(raw_example_i)
             elif _is_record_data(dataset_hparams[i]["data_type"]):
-                example: Dict[str, Any] = raw_example_i  # type: ignore
-                # for some reasons even after the not None check here, mypy
-                # thinks self._dataset_features[i] may be None, hence adding
-                # ignore to the lines below
-                if self._dataset_features[i] is not None:
-                    for key, dtype in \
-                            self._dataset_features[i]["convert_types"].items():  # type: ignore # pylint: disable=line-too-long
-                        example[key] = np.asarray(example[key], dtype=dtype)
-                    for key, transform in \
-                            self._dataset_features[i]["image_transforms"].items():  # type: ignore # pylint: disable=line-too-long
-                        example[key] = transform(example[key])
-                for transform in self._other_transforms[i]:
-                    example = transform(example)
-                processed_examples.append(example)
+                example_i = self._databases[i].process(raw_example_i)
+            for transform in self._other_transforms[i]:
```

Comment:
I don't think we need to deal with `other_transforms` anymore. They should be dealt with by individual data bases.

Reply:
Resolved in 462312cab452a7f1c3eacce634d65f1a50c2fbbc